src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs

```diff
@@ -0,0 +1,230 @@
+                var fullPath = openApiDocument.Servers[0].Url;
+                var uri = new Uri(fullPath);
+                basePath = uri.PathAndQuery;
+            }
+
+            return basePath;
+        }
+
+        // Get suggested options values.  Returns null if none. 
+        public static string[] GetOptions(this OpenApiParameter param)
+        {
+            // x-ms-enum-values is: array of { value :string, displayName:string}.
+            if (param.Extensions.TryGetValue("x-ms-enum-values", out var value))
+            { 
+                if (value is OpenApiArray array)
+                {
+                    var list = new List<string>(array.Capacity);
+
+                    foreach (var item in array)
+                    {
+                        if (item is OpenApiObject obj)
+                        {
+                            // has keys, "value", and "displayName"
+                            if (obj.TryGetValue("value", out var value2))
+                            {
+                                if (value2 is OpenApiString str)
+                                {
+                                    list.Add(str.Value);
+                                    continue;
+                                }
+                            }
+                        }
+
+                        throw new NotImplementedException($"Unrecognized x-ms-enum-values schema");
+                    }
+
+                    return list.ToArray();
+                }
+            }
+
+            return null;
+        }
+
+        public static bool IsTrigger(this OpenApiOperation op)
+        {
+            var isTrigger = op.Extensions.ContainsKey("x-ms-trigger");
+            return isTrigger;
+        }
+
+        public static bool TryGetDefaultValue(this OpenApiParameter param, out string defaultValue)
+        {
+            var x = param.Schema.Default;
+            if (x == null)
+            {
+                defaultValue = null;
+                return false;
+            }
+
+            if (x is OpenApiString str)
+            {
+                defaultValue = str.Value;
+                return true;
+            }
+
+            if (x is OpenApiInteger intVal)
+            {
+                defaultValue = intVal.Value.ToString();
+                return true;
+            }
+
+            if (x is OpenApiDouble dbl)
+            {
+                defaultValue = dbl.Value.ToString();
+                return true;
+            }
+
+            if (x is OpenApiBoolean b)
+            {
+                defaultValue = b.Value.ToString();
+                return true;
+            }
+
+            throw new NotImplementedException($"Unknown default value type {x.GetType().FullName}");
+        }
+
+        public static bool HasDefaultValue(this OpenApiParameter param)
+        {
+            return param.Schema.Default != null;
+        }
+
+        public static bool IsInternal(this OpenApiParameter param)
```

Comment:
I'd prefer the shorter version `public static bool IsInternal(this OpenApiParameter param) => param.Extensions.TryGetValue("x-ms-visibility", out _);`